tyr-runner/src/main/java/org/jboss/tyr/ci/TeamCityCI.java

```diff
@@ -101,29 +124,100 @@ public void triggerBuild(JsonObject prPayload) {
+        setTriggerBuildProperties(prPayload);
+        if (!branchMappings.containsKey(branch)) {
+            return;
+        }
+        setSnapshotDependencies(buildId, true);
+        triggerBuild(branch, pull, sha, buildId);
     }
 
-    private String getBaseUrl(String host, int port) {
-        return String.format("http://%s:%d/httpAuth", host, port);
+    private void setSnapshotDependencies(String buildId, boolean retestOnlyFailed) {
+        SnapshotDependencies snapshotDependencies = getSnapshotDependencies(buildId);
+
+        if (snapshotDependencies.snapshotDependencyCount.intValue() < 1) {
+            log.info("No snapshot dependency.");
+            return;
+        }
+
+        for (SnapshotDependency dependency: snapshotDependencies.snapshotDependencies) {
+            for (Property prop :dependency.properties.propertyList) {
+                if (prop.name.equals("take-started-build-with-same-revisions") || prop.name.equals("take-successful-builds-only")) {
+                    prop.value = String.valueOf(retestOnlyFailed);
+                }
+            }
+        }
+
+        Jsonb jsonb = JsonbBuilder.create();
+        String stringJsonb = jsonb.toJson(snapshotDependencies);
+        Entity<String> json = Entity.json(stringJsonb);
+
+        String path = "app/rest/buildTypes/id:" + buildId + "/snapshot-dependencies";
+        WebTarget target = returnTeamCityTarget(path);
+        Response response = null;
+
+        try {
+            response = target.request()
+                .header(HttpHeaders.ACCEPT_ENCODING, "UTF-8")
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+                .header(HttpHeaders.AUTHORIZATION, "Basic " + encryptedCredentials)
+                .put(json);
+
+        } catch (Throwable e) {
+            log.error("Cannot retrieve JSON on the path " + path, e);
+            throw e;
+        } finally {
+            if (response != null) {
+                response.close();
+            }
+        }
     }
 
-    private Map<String, String> parseBranchMapping(String mappings) {
-        Map<String, String> branchMap = new HashMap<>();
+    private SnapshotDependencies getSnapshotDependencies(String buildId) {
+        String path = "app/rest/buildTypes/id:" + buildId + "/snapshot-dependencies";
+        WebTarget target = returnTeamCityTarget(path);
+        Response response = null;
+        SnapshotDependencies snapshotDependencies;
 
-        for (String mapping : mappings.split(",")) {
-            String[] parts = mapping.split("=>");
-            branchMap.put(parts[0].trim(), parts[1].trim());
+        try {
+            response = target.request()
+                .header(HttpHeaders.ACCEPT_ENCODING, "UTF-8")
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+                .header(HttpHeaders.AUTHORIZATION, "Basic " + encryptedCredentials)
+                .header(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON)
```

Comment:
maybe you can return somewhere this whole request builder object so you don't need to repeat all the headers everytime?

Reply:
or even more generic `getJson(path)`